services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/MultipartDownloaderSubscriber.java

```diff
@@ -113,11 +114,13 @@ public void onNext(AsyncResponseTransformer<GetObjectResponse, GetObjectResponse
             throw new NullPointerException("onNext must not be called with null asyncResponseTransformer");
         }
 
-        int nextPartToGet = completedParts.get() + 1;
+        int currentPart = completedParts.get();
+        int nextPartToGet =  currentPart + 1;
 
         synchronized (lock) {
             if (totalParts != null && nextPartToGet > totalParts) {
                 log.debug(() -> String.format("Completing multipart download after a total of %d parts downloaded.", totalParts));
+                validatePartsCount(currentPart);
```

Comment:
I believe we should remove validation from here because it's totally valid for publisher to send more data after subscription. 

https://github.com/reactive-streams/reactive-streams-jvm

> 8 	If a Subscription is cancelled its Subscriber MUST eventually stop being signaled.
:bulb: 	The intent of this rule is to make sure that Publishers respect a Subscriber’s request to cancel a Subscription when Subscription.cancel() has been called. The reason for eventually is because signals can have propagation delay due to being asynchronous.